### ISSUE TITLE
Version not compatible with IntelliJ 2023.3.x

### ISSUE THREAD

**Issue by adrian-goe** [KronicDeth/intellij-elixir#3481]
Hi, could you update the build to be compatible with the latest IntelliJ version? Thanks 🙏🏻 

![Screenshot 2023-12-13 at 14 01 10](https://github.com/KronicDeth/intellij-elixir/assets/17064122/5557f90e-eacb-4aed-8444-334070be641a)
![Screenshot 2023-12-13 at 14 00 40](https://github.com/KronicDeth/intellij-elixir/assets/17064122/100f7993-3754-4fd7-a5fd-d41636aec653)

**Comment by mengelseatz** [comment]
It looks like there was already code committed/merged for this?   I don't know what the process is to get this into JetBrains though 

https://github.com/KronicDeth/intellij-elixir/commit/876b7702e439c398f4ae2cc41647eb45d2a157ea

**Comment by adrian-goe** [comment]
@mengelseatz yes, it seems to be available in the latest canary version. If that is stable enough, release it 😄 

But as a side note, I noticed something:
According to the [canary changelogs for version 16.0.1](https://plugins.jetbrains.com/plugin/7522-elixir/versions/stable/394979), its available in version 16.0.0.
![Screenshot 2023-12-17 at 14 50 15](https://github.com/KronicDeth/intellij-elixir/assets/17064122/7a738199-77ec-4ebd-a77a-598661249f15)

According to latest [stable build](https://plugins.jetbrains.com/plugin/7522-elixir) build on the plugin page, it says only support Intelij 2023.2 
![Screenshot 2023-12-17 at 14 50 54](https://github.com/KronicDeth/intellij-elixir/assets/17064122/6b13f1ba-2b86-4dbe-a1ce-240dcb6651a0)

**Comment by mengelseatz** [comment]
Weird.. clearly it's not working in 16.0.0  though 😄. 
I guess we need to wait for a build at this point.  I'm not familiar with the build cycle for this project as I just got back into Elixir a few weeks ago after a multi-year hiatus.    Luckily Emacs works well enough for me to continue working.

**Comment by MontiniCristian** [comment]
@mengelseatz you can download the latest tag (pre-release) as a zip file and manually install it, it works smoothly on my machine with RubyMine 2023.3.1:

![Screenshot from 2023-12-19 09-01-28](https://github.com/KronicDeth/intellij-elixir/assets/23277928/d1ac8109-b293-4786-8817-df21283a608d)

**Comment by mengelseatz** [comment]
Thanks @MontiniCristian !

**Comment by Cervajz** [comment]
I think it was not released because the tests failed on the latest merge:

https://github.com/KronicDeth/intellij-elixir/actions/runs/7147009230

**Comment by KronicDeth** [comment]
Yep, the tests failed, so I need to go in and fix the compatibility problems

**Comment by piacsek** [comment]
> Yep, the tests failed, so I need to go in and fix the compatibility problems

Hi KronicDeth! I really like Elixir, Idea and this plugin. I'd like to start contributing here. I see there many open issues, could you point me a good one to get started? 

Thanks for putting so much time and effort into this plugin, BTW!

**Comment by piacsek** [comment]
@MontiniCristian It didn't work for me :/ Getting the same error for all 16.X.X versions

![image](https://github.com/KronicDeth/intellij-elixir/assets/9383701/85dd56e7-753e-4a05-8c0b-7ce18f08a29b)

**Comment by Cervajz** [comment]
@piacsek Did you download the correct file from the Canary channel? The name should be _intellij-elixir-16.0.1-pre_20231208222626.zip_, your screenshot shows 16.0.**0** 

https://plugins.jetbrains.com/plugin/7522-elixir/versions/canary

**Comment by derekcroft** [comment]
@MontiniCristian I downloaded the file you mentioned and got the same message as @piacsek except with the newer version:

![Screenshot 2024-01-05 at 5 37 56 PM](https://github.com/KronicDeth/intellij-elixir/assets/62440/40efc5de-8051-43e6-9fb1-c05aa6f7c670)

I uninstalled the previous version of the plugin and restarted the IDE, but no luck.

**Comment by amarchen** [comment]
Seems to be working now in 2023.3.3. Well, debugging still fails with "Connection attempt from node... since it cannot handle [~c"V4_NC", ~c"UNLINK_ID"]", but that's another issue - https://github.com/KronicDeth/intellij-elixir/issues/3362

**Comment by sh41** [comment]
This was fixed a long time ago, but the thread never got a straight answer, so here is the whole timeline.

**What went wrong.** v16.0.0 (published 12 Sep 2023) declared an upper compatibility bound of `until-build 232.*` — IntelliJ 2023.2 and older. When 2023.3 shipped as build `233.*`, the IDE refused to load the plugin and showed exactly the banner in adrian-goe's screenshot: *"requires IDE build 232.\* or earlier"*. No plugin code ever ran; this was purely the declared version range being behind the IDE.

**The fix landed before the issue was filed.** Commit 876b7702e, "Support 233 (2023.3 release)", raised the ceiling to `233.*` on 14 Oct 2023 — two months before this report. It then sat unreleased for three months because the same 2023.3 upgrade broke the test suite and the release was held on a red build. That is what mengelseatz found on `main` and what the maintainer meant by *"the tests failed, so I need to go in and fix the compatibility problems"*. The CI run linked in this thread has since aged past GitHub's log retention, so the actual failure can no longer be read — we can say it blocked the release, not why.

**The release that fixed it is v17.0.0, published 11 Jan 2024**, declaring `since-build 233.11799.241` / `until-build 233.*`. amarchen confirmed here three weeks later, and #3494 — the same problem reported separately — was closed by its reporter for the same reason.

**Why this specific failure cannot happen again.** It was not a one-off: the same ceiling refused 2023.2 (#3325, #3326, #3343), then 2023.3 (this issue), then 2024.1 (#3566, #3578 and #3620), once per IDE cycle. That stopped because the upper bound was dropped entirely, not just raised again. Published upper bounds, from the Marketplace:

| Plugin version | Published | Declared `until-build` |
|---|---|---|
| 16.0.0 | 2023-09-12 | `232.*` ← what this issue hit |
| 17.0.0 | 2024-01-11 | `233.*` ← the fix |
| 18.0.1 / 19.x / 20.x | 2024 | `241.*` / `242.*` / `243.*` |
| 21.0.0 | 2025-05-17 | none |
| 22.0.1 … 24.0.1 | 2025-12-16 … 2026-08-09 | none |

Since v21.0.0 the plugin declares no upper bound at all, and it still doesn't in the current v24.0.1. There is no ceiling left to fall behind a future IDE, so an install can no longer be refused this way. The commit that dropped it says why in as many words: *"We want users to be able to install the plugin on future versions, and if there is incompatibility, they should hopefully create an issue :-)."* So if a future IDE does break something, the plugin will load and fail visibly rather than refuse to install.

**The "Fail to load plugin descriptor" errors in this thread are a different problem.** piacsek and derekcroft hit those while sideloading the 16.0.x pre-release zips as a workaround. That is a packaging fault in those particular canary builds, not the compatibility range, and it is not tracked by any issue. If you still see it on a current release, please open a new issue with the IDE version and the exact zip filename — it will not get fixed inside this one.

Closing as completed: fixed in v17.0.0, and structurally prevented since v21.0.0.

Closed as part of an AI-assisted review of the backlog. A human checked this one before closing it, but it might still be a mistake. If you think the reason above is wrong, please reopen the issue and tell us why.
